### DISCUSSION TITLE
Inline image attachments

### DISCUSSION THREAD

**Discussion by WarlockFish** [standardnotes/forum#2825] [Suggestions] [answered]
It is very important to insert uploaded pictures and files into the text. I hope it can be supported as soon as possible。

when I write a markdown ，I need a photo，standardnotes can be uploaded, but cannot be used

**Comment by adamhotep** [comment]
To my understanding, that functionality is on the current roadmap, as noted at [How do I attach encrypted files to my notes?](https://standardnotes.com/help/36/how-do-i-attach-encrypted-files-to-my-notes):

>Files are presently an "outside-of-editor" experience, meaning that encrypted photos and videos cannot be embedded directly inside of a note. The ability to embed photos inside of editors is on our feature roadmap. 

and as (less directly) noted in the March 2022 [Roadmap Update](https://blog.standardnotes.com/33124/roadmap-update):
> Built-in encrypted files. Possibly one of the most important features we’ve worked on to date. We’re making excellent progress here, particularly on the backend, infrastructure, and client libraries.

See also standardnotes/forum#96, which tracks this request (but hasn't been updated since 2019, before the encrypted file work began).

This feature was mentioned in the "we don't have a [public] roadmap" discussion at standardnotes/forum#667 on 2021-03-02, when @JaspalSuri noted that "the dev teams plans to work on ... The ability to natively upload files and upload images within the apps", so it's already been somewhat prioritized for two years. Of course, the same post also noted that "we can't predict when certain features will arrive".

**Comment by bharker75** [comment]
Any updates on the inline embedded images feature? Love the SN product, and I can work around this limitation by self hosting images and linking via URL, but seamless embedded images would be a huge improvement to the user experience.

**Comment by effieeee** [comment, marked as answer]
Hi @bharker75, there's currently an experimental editor with image embedding, though it's only accessible to Professional Plan holders at the moment!

![image](https://user-images.githubusercontent.com/54989748/203340648-27f9c354-c9c9-4fa5-ac27-f9a2a9505e05.png)

**Reply by webysther** [reply]
No way this is a good answer, super don't allow paste richtext or markdown and works terrible when moving notes around, vendor lock-in.

**Reply by effieeee** [reply]
Hi @webysther, the original poster's question pertains to whether or not there's a note type that supports image embedding in particular, which the Super note type is very much capable of.

If you're currently experiencing some issues with the formatting of pasted content on the Super editor, please open a separate [issue](https://github.com/standardnotes/forum/issues)! 🙏

**Comment by bharker75** [comment]
@effieeee I have a Professional Plan so that's great news. Thanks!
